src/keywrap/keyprovider.rs

```diff
@@ -334,8 +334,8 @@ impl KeyProviderKeyWrapper {
             });
             match handler.join() {
                 Ok(Ok(v)) => Ok(v),
-                Ok(Err(e)) => return Err(anyhow!("failed to unwrap key by gRPC, {e}")),
-                Err(e) => return Err(anyhow!("failed to unwrap key by gRPC, {e:?}")),
+                Ok(Err(e)) => bail!("failed to unwrap key by gRPC, {e}"),
```

Comment:
No need for `bail!`, just `Err(anyhow!("failed to unwrap key by gRPC, {e}"))` should work.

Reply:
Yes, but the `bail` macro is preferred because it can reduce a lot of repetition of`Error(anyhow!())` s. I think it is more concise